RealmSwift/AnyRealmValue.swift

```diff
@@ -0,0 +1,150 @@
+    case string(String)
+    /// A binary data type.
+    case data(Data)
+    /// A date type.
+    case date(Date)
+    /// A Realm Object type.
+    case object(Object)
+    /// An ObjectId type.
+    case objectId(ObjectId)
+    /// A Decimal128 type.
+    case decimal128(Decimal128)
+    /// A UUID type.
+    case uuid(UUID)
+
+    /// Returns an `Int` if that is what the stored value is, otherwise `nil`.
+    public var intValue: Int? {
+        guard case let .int(i) = self else {
+            return nil
+        }
+        return i
+    }
+
+    /// Returns a `Bool` if that is what the stored value is, otherwise `nil`.
+    public var boolValue: Bool? {
+        guard case let .bool(b) = self else {
+            return nil
+        }
+        return b
+    }
+
+    /// Returns a `Float` if that is what the stored value is, otherwise `nil`.
+    public var floatValue: Float? {
+        guard case let .float(f) = self else {
+            return nil
+        }
+        return f
+    }
+
+    /// Returns a `Double` if that is what the stored value is, otherwise `nil`.
+    public var doubleValue: Double? {
+        guard case let .double(d) = self else {
+            return nil
+        }
+        return d
+    }
+
+    /// Returns a `String` if that is what the stored value is, otherwise `nil`.
+    public var stringValue: String? {
+        guard case let .string(s) = self else {
+            return nil
+        }
+        return s
+    }
+
+    /// Returns `Data` if that is what the stored value is, otherwise `nil`.
+    public var dataValue: Data? {
+        guard case let .data(d) = self else {
+            return nil
+        }
+        return d
+    }
+
+    /// Returns a `Date` if that is what the stored value is, otherwise `nil`.
+    public var dateValue: Date? {
+        guard case let .date(d) = self else {
+            return nil
+        }
+        return d
+    }
+
+    /// Returns an `ObjectId` if that is what the stored value is, otherwise `nil`.
+    public var objectIdValue: ObjectId? {
+        guard case let .objectId(o) = self else {
+            return nil
+        }
+        return o
+    }
+
+    /// Returns a `Decimal128` if that is what the stored value is, otherwise `nil`.
+    public var decimal128Value: Decimal128? {
+        guard case let .decimal128(d) = self else {
+            return nil
+        }
+        return d
+    }
+
+    /// Returns a `UUID` if that is what the stored value is, otherwise `nil`.
+    public var uuidValue: UUID? {
+        guard case let .uuid(u) = self else {
+            return nil
+        }
+        return u
+    }
+
+    /// Returns the stored value as a Realm Object of a specific type.
+    ///
+    /// - Parameter objectType: The type of the Object to return.
+    /// - Returns: A Realm Object of the supplied type if that is what the underlying value is,
+    /// otherwise `nil` is returned.
+    public func objectValue<T: Object>(_ objectType: T.Type) -> T? {
```

Comment:
Might want to consider changing the name slightly to reflect that this is a function, not a variable.

Also, based on @nirinchev points, we should also expose a method that returns the value as a `DynamicObject`:

```swift
public fun dynamicObject() -> DynamicObject?
```